File: app/src/main/java/pk89/data/Presets.kt
package pk89.data

/** Справочное значение г/ч.л. (чайная ложка без горки, приблизительно). Коэффициент можно править в рецепте. */
data class IngredientPreset(val name: String, val gramsPerTeaspoon: Double)

object Presets {
    val all: List<IngredientPreset> = listOf(
        IngredientPreset("Сахар", 5.0),
        IngredientPreset("Соль", 7.0),
        IngredientPreset("Крахмал картофельный", 6.0),
        IngredientPreset("Крахмал кукурузный", 5.0),
        IngredientPreset("Мука пшеничная", 4.0),
        IngredientPreset("Мука гороховая", 4.0),
        IngredientPreset("Горох колотый молотый", 5.0),
        IngredientPreset("Сухое молоко", 5.0),
        IngredientPreset("Сухие сливки", 4.0),
        IngredientPreset("Какао", 4.0),
        IngredientPreset("Лимонная кислота", 5.0),
        IngredientPreset("Ягодный порошок", 3.0),
        IngredientPreset("Сушёные овощи", 2.0),
        IngredientPreset("Сушёный лук", 2.0),
        IngredientPreset("Сушёная морковь", 2.0),
        IngredientPreset("Паприка", 3.0),
        IngredientPreset("Перец чёрный молотый", 3.0),
        IngredientPreset("Сухая зелень", 1.0),
        IngredientPreset("Бульонная основа", 5.0),
        IngredientPreset("Ванилин", 4.0),
    )

    fun find(name: String): IngredientPreset? =
        all.firstOrNull { it.name.equals(name.trim(), ignoreCase = true) }
}

File: js/presets.js
// Справочные г/ч. л. — чайная ложка без горки, приблизительно.
// Коэффициент правится прямо в рецепте, здесь только стартовое значение.

export const PRESETS = [
  { name: 'Сахар', gramsPerTeaspoon: 5 },
  { name: 'Соль', gramsPerTeaspoon: 7 },
  { name: 'Крахмал картофельный', gramsPerTeaspoon: 6 },
  { name: 'Крахмал кукурузный', gramsPerTeaspoon: 5 },
  { name: 'Мука пшеничная', gramsPerTeaspoon: 4 },
  { name: 'Мука гороховая', gramsPerTeaspoon: 4 },
  { name: 'Горох колотый молотый', gramsPerTeaspoon: 5 },
  { name: 'Сухое молоко', gramsPerTeaspoon: 5 },
  { name: 'Сухие сливки', gramsPerTeaspoon: 4 },
  { name: 'Какао', gramsPerTeaspoon: 4 },
  { name: 'Лимонная кислота', gramsPerTeaspoon: 5 },
  { name: 'Ягодный порошок', gramsPerTeaspoon: 3 },
  { name: 'Сушёные овощи', gramsPerTeaspoon: 2 },
  { name: 'Сушёный лук', gramsPerTeaspoon: 2 },
  { name: 'Сушёная морковь', gramsPerTeaspoon: 2 },
  { name: 'Паприка', gramsPerTeaspoon: 3 },
  { name: 'Перец чёрный молотый', gramsPerTeaspoon: 3 },
  { name: 'Сухая зелень', gramsPerTeaspoon: 1 },
  { name: 'Бульонная основа', gramsPerTeaspoon: 5 },
  { name: 'Ванилин', gramsPerTeaspoon: 4 },
];

export const DEFAULT_GRAMS_PER_TSP = 5;

export const EMOJIS = ['🥣', '🫐', '🍒', '🍓', '🍎', '🌾', '🥔', '🧅', '🥕', '🌽', '🍲', '☕', '🍵', '🍫', '🥛', '🧂'];

export function findPreset(name) {
  const needle = (name || '').trim().toLowerCase();
  return PRESETS.find((p) => p.name.toLowerCase() === needle) || null;
}
